test/wrapped-token.js

```diff
@@ -0,0 +1,105 @@
+/* globals artifacts */
+
+import BN from "bn.js";
+import chai from "chai";
+import bnChai from "bn-chai";
+
+import { checkErrorRevert } from "../helpers/test-helper";
+
+const { expect } = chai;
+chai.use(bnChai(web3.utils.BN));
+
+const Token = artifacts.require("Token");
+const TokenAuthority = artifacts.require("TokenAuthority");
+const WrappedToken = artifacts.require("WrappedToken");
+
+contract("Wrapped Token", accounts => {
+  const USER0 = accounts[0];
+  const USER1 = accounts[1];
+
+  const WAD = new BN(10).pow(new BN(18));
+  const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+  let token;
+  let wrappedToken;
+
+  beforeEach(async () => {
+    token = await Token.new("Colony token", "CLNY", 18);
+    wrappedToken = await WrappedToken.new(token.address);
+
+    const tokenAuthority = await TokenAuthority.new(token.address, ZERO_ADDRESS, [wrappedToken.address]);
+    await token.setAuthority(tokenAuthority.address);
```

Comment:
This isn't really representative of the situation we're going to be using this contract in on Xdai. On the Xdai side, `token` is a pure-ish ERC20 contract (well, I think strictly an ERC677 contract). There is no authority we're able to lean on for permissions.

Reply:
Yeah fair!